qiskit/_compiler.py

```diff
@@ -1,5 +1,7 @@
 # -*- coding: utf-8 -*-
 # pylint: disable=redefined-builtin
+# pylint: disable=C0103
+# pylint: disable=W9006
```

Comment:
Can you remove this pylint exceptions and take care of the warning themselves instead? In general, disabling a linter check (specially in `qiskit/*`) should only be used as a last resort, not as a convenience, in order to maintain the consistency of the codebase.

Reply:
Fixed the raising exception issue.  Not going to worry about how my iterator constants are defined.